### PR TITLE
Corrects the call to generate events that were commented. Tests again…

### PR DESCRIPTION
**Pull request by glcj**
Corrects call to generate asynchronous events for the S7 driver.

### CODE REVIEW

**Comment by chrisdutz** [comment]
Had a look and it seems all changes only have an effect on the Event-based PLCs ... none of which I have. So I have no objections to merging this as is ;-)

File: plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java [hunk 1]
```diff
@@ -1448,54 +1452,86 @@ private CompletableFuture<S7Message> sendInternal(S7Message request) {
...
      */
     @Override
     protected void decode(ConversationContext<TPKTPacket> context, TPKTPacket msg) throws Exception {
-
-        S7Message s7msg = msg.getPayload().getPayload();
-        S7Parameter parameter = s7msg.getParameter();
-        if (parameter instanceof S7ParameterModeTransition) {
-            // TODO: The eventQueue is only drained in the S7ProtocolEventLogic.ObjectProcessor and here only messages of type S7Event are processed, so S7PayloadUserDataItem elements will just be ignored.
-            //eventQueue.add(parameter);
+        System.out.println(msg);
```
Comment (sruehl) [inline plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java]: maybe remove that before merge

File: plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java [hunk 2]
```diff
@@ -1448,54 +1452,86 @@ private CompletableFuture<S7Message> sendInternal(S7Message request) {
...
+            
         } else if (parameter instanceof S7ParameterUserData) {
+            
             S7ParameterUserData parameterUD = (S7ParameterUserData) parameter;
             List<S7ParameterUserDataItem> parameterUDItems = parameterUD.getItems();
+            
             for (S7ParameterUserDataItem parameterUDItem : parameterUDItems) {
+                
                 if (parameterUDItem instanceof S7ParameterUserDataItemCPUFunctions) {
+
```
Comment (sruehl) [inline plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java:1493]: why are there so many empty lines now?

File: plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java [hunk 3]
```diff
@@ -1448,54 +1452,86 @@ private CompletableFuture<S7Message> sendInternal(S7Message request) {
...
                             (myParameter.getCpuSubfunction() == 0x13) ||
-                            (myParameter.getCpuSubfunction() == 0x16))) {
-                        S7PayloadUserData payload = (S7PayloadUserData) s7msg.getPayload();
-                        List<S7PayloadUserDataItem> items = payload.getItems();
-                        // TODO: The eventQueue is only drained in the S7ProtocolEventLogic.ObjectProcessor and here only messages of type S7Event are processed, so S7PayloadUserDataItem elements will just be ignored.
-                        //eventQueue.addAll(items);
+                            (myParameter.getCpuSubfunction() == 0x16))) { //(04)
+                        
+                        payload.getItems().forEach(item ->{
+                            //if (item instanceof S7PayloadDiagnosticMessage) {
```
Comment (sruehl) [inline plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java]: is it supposed to be outcommented?

Comment (glcj) [reply plc4j/drivers/s7/src/main/java/org/apache/plc4x/java/s7/readwrite/protocol/S7ProtocolLogic.java]: Hello, thanks for the comments

1. println, was removed.
2. The separated lines are for my better reading of the code.
3. The commented lines were removed.
4. Added "TODO:" in code sections that require reverse engineering.

Grateful,

**Pull request merged**